File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof24.vb
Public Module Example
    Public Sub Main()
        ' <Snippet24>
        Dim searchString As String = ChrW(&HAD) + "m"

        Dim s1 As String = "ani" + ChrW(&HAD) + "m"
        Dim s2 As String = "animal"

        Dim position As Integer

        position = s1.LastIndexOf("m"c)
        If position >= 1 Then
            Console.WriteLine(s1.LastIndexOf(searchString, position, position, StringComparison.CurrentCulture))
            Console.WriteLine(s1.LastIndexOf(searchString, position, position, StringComparison.Ordinal))
        End If

        position = s2.LastIndexOf("m"c)
        If position >= 1 Then
            Console.WriteLine(s2.LastIndexOf(searchString, position, position, StringComparison.CurrentCulture))
            Console.WriteLine(s2.LastIndexOf(searchString, position, position, StringComparison.Ordinal))
        End If

        ' The example displays the following output:
        '
        ' 4
        ' 3
        ' 3
        ' -1
        ' </Snippet24>
    End Sub
End Module

File: snippets/fsharp/System/String/LastIndexOf/lastindexof24.fs
module lastindexof24.fs
open System

do
    // <Snippet24>
    let searchString = "\u00ADm"

    let s1 = "ani\u00ADmal"
    let s2 = "animal"

    let position = s1.LastIndexOf 'm'
    if position >= 1 then
        printfn $"{s1.LastIndexOf(searchString, position, position, StringComparison.CurrentCulture)}"
        printfn $"{s1.LastIndexOf(searchString, position, position, StringComparison.Ordinal)}"

    let position = s2.LastIndexOf 'm'
    if position >= 1 then
        printfn $"{s2.LastIndexOf(searchString, position, position, StringComparison.CurrentCulture)}"
        printfn $"{s2.LastIndexOf(searchString, position, position, StringComparison.Ordinal)}"

    // The example displays the following output:
    //
    // 4
    // 3
    // 3
    // -1
    // </Snippet24>
